### PR TITLE
Add RESP3 migration guide

### PR DESCRIPTION
**Pull request by uglide**
Add guide for Migrating from RESP2 to RESP3 Replies

CC: @chayim @leibale @dwdougherty

### CODE REVIEW

**Comment by chayim** [comment]
@uglide does this get published to redis.io at all via the redis-stack-docs... or should we cross-link?

File: docs/docs/resp3.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+description: JSON RESP2 to RESP3 replies reference for clients developers
+---
+
+In RESP3, the default value of the optional path argument was changed from `.` to `$`. 
+Due to this change, the replies of some commands have slightly changed. 
+This page provides a brief comparison between RESP2 and RESP3 responses for JSON commands to help developers in migrating their clients from RESP2 to RESP3.
+
+### JSON Command Replies Comparison 
+
+The types are described using a [“TypeScript-like” syntax](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html). `Array<a>` denotes an [array](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#arrays) where the type of elements is known, but the number of elements is not.
```
Comment (chayim) [inline docs/docs/resp3.md:14]: Can we show a couple of before/after query examples?

**Comment by uglide** [comment]
@chayim redis.io build will automatically pick up the file 
Here is a demo https://uglide-demo-redis-docs.netlify.app/docs/data-types/json/resp3/

File: docs/docs/resp3.md (hunk 1 above)
Comment (uglide) [reply docs/docs/resp3.md:14]: I've updated the table and added example for the `JSON.GET`. All other commands seem to have minor response changes like BulkString -> number.

File: docs/docs/resp3.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+title: "Guide for Migrating from RESP2 to RESP3 Replies"
+linkTitle: "RESP3 migration guide"
+weight: 6
+description: JSON RESP2 to RESP3 replies reference for clients developers
```
Comment (dwdougherty) [inline docs/docs/resp3.md:5]:
```suggestion
description: JSON RESP2 to RESP3 replies reference for client developers
```

File: docs/docs/resp3.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+---
+
+In RESP3, the default value of the optional path argument was changed from `.` to `$`. 
+Due to this change, the replies of some commands have slightly changed. 
+This page provides a brief comparison between RESP2 and RESP3 responses for JSON commands to help developers in migrating their clients from RESP2 to RESP3.
+
+### JSON Command Replies Comparison
```
Comment (dwdougherty) [inline docs/docs/resp3.md:12]: Style change.

```suggestion
### JSON command replies comparison 
```

File: docs/docs/resp3.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+---
+title: "Guide for Migrating from RESP2 to RESP3 Replies"
```
Comment (dwdougherty) [inline docs/docs/resp3.md:2]:
```suggestion
title: "Guide for migrating from RESP2 to RESP3 replies"
```

**Review by dwdougherty — CHANGES_REQUESTED** [review]
just a few style changes

**Comment by chayim** [comment]
@uglide merge in the style changes + merge.. Approved pending.

**Review by chayim — APPROVED** [review]

**Review by chayim — APPROVED** [review]

**Comment by uglide** [comment]
@chayim Thanks!

**Pull request merged**